### PR TITLE
Scheduler API cleanup

### PR DESCRIPTION
**Pull request by andyross**
Getting started looking at the scheduler.  Step one is some work to reduce the external API footprint in the rest of the kernel, which was a little cluttered.  In theory all of this is just cleanup with at most minimal change to generated code.  No logic changes.

### CODE REVIEW

File: kernel/queue.c [hunk 1]
```diff
@@ -148,26 +148,21 @@ void k_queue_append_list(struct k_queue *queue, void *head, void *tail)
...
 	__ASSERT(head && tail, "invalid head or tail");
 
-	unsigned int key = irq_lock();
+	int need_sched = 0;
+	unsigned int was_nonempty, key = irq_lock();
 #if !defined(CONFIG_POLL)
-	struct k_thread *first_thread, *thread;
+	struct k_thread *thread;
 
-	first_thread = _peek_first_pending_thread(&queue->wait_q);
```
Comment (lpereira) [inline kernel/queue.c]: Maybe get rid of this function now that nothing else calls it?

Comment (andyross) [reply kernel/queue.c]: That will come.  There are actually a *lot* of dead functions in ksched.h.  I think the way this is evolving is that I'll end up writing a new header and adding back only what is needed.

**Comment by andyross** [comment]
Had to add one more patch because the change that added the event logger hook to all thread ready operations also added it to the initialization of the main and idle threads which happen before the logging system is initialized.

File: kernel/include/ksched.h [hunk 2]
```diff
@@ -541,4 +526,10 @@ static inline int _is_thread_user(void)
 #endif
 }
 #endif /* CONFIG_USERSPACE */
+
+/**
```
Comment (lpereira) [inline kernel/include/ksched.h]: `#if defined(CONFIG_USE_SWITCH)`?

File: subsys/logging/kernel_event_logger.c [hunk 3]
```diff
@@ -47,6 +49,8 @@ static int _sys_k_event_logger_init(struct device *arg)
 	sys_event_logger_init(&sys_k_event_logger, _sys_k_event_logger_buffer,
 		CONFIG_KERNEL_EVENT_LOGGER_BUFFER_SIZE);
 
+	initialized = 1;
```
Comment (lpereira) [inline subsys/logging/kernel_event_logger.c]: Good catch.

**Review by lpereira — APPROVED** [review]
Patch LGTM; it doesn't look like it'll introduce any regressions.

**Review by nashif — APPROVED** [review]

**Pull request merged**